app/publisher/publisher.py

```diff
@@ -0,0 +1,51 @@
+from abc import ABC, abstractmethod
+
+import google
+from google.cloud.pubsub_v1 import PublisherClient
+from google.cloud.pubsub_v1.futures import Future
+from structlog import get_logger
+
+from app.publisher.publication_failed import PublicationFailed
+
+logger = get_logger(__name__)
+
+
+class Publisher(ABC):
+    @abstractmethod
+    def publish(self, topic_id, message):
+        pass  # pragma: no cover
+
+
+class PubSubPublisher(Publisher):
+    def __init__(self):
+        self._client = PublisherClient()
+        _, self._project_id = google.auth.default()
+
+    def _publish(self, topic_id, message):
+        logger.info("publishing message", topic_id=topic_id)
+        topic_path = self._client.topic_path(self._project_id, topic_id)
+        publish_future: Future = self._client.publish(topic_path, message)
```

Comment:
This var should be something like `response`, as it doesn't matter it's a future as the future will get resolved

Reply:
It's not really a response, I would say it becomes a response when you resolve it.

Reply:
Google disagrees https://googleapis.dev/python/pubsub/latest/publisher/api/client.html 😆

Reply:
As discussed, changed to keep it inline with official library,